### DISCUSSION TITLE
Auto Defragmentation after Auto Compaction

### DISCUSSION THREAD

**Discussion by allahshukur-ahmadzada** [etcd-io/etcd#15643] [Q&A] [answered]
I see there is env variable to schedule auto compaction but cant find one for auto defragmentation. How important to have defragmentation after auto compaction?

**Comment by vwalek** [comment, marked as answer]
Hi, I don't think there is auto defrag implemented yet, per docs. 

> How important to have defragmentation after auto compaction?
IMHO, it depends on the size of the database. Having size larger than >1G, then it should be performed.

> The process of defragmentation releases this storage space back to the file system. 
> https://etcd.io/docs/v3.4/op-guide/maintenance/

**Reply by jmhbnz** [reply]
Hey @vwalek - Thanks for your help answering this discussion. You are absolutely right there is no auto defragmentation feature currently, though it is still an important part of etcd maintenance as you mention to ensure enough backend storage space is available. There are community tools for example https://github.com/ahrtr/etcd-defrag which is intended to make the maintenance process of defrag easier for cluster operators.
